### PR TITLE
FEATURE: Clean up `PostReplyKey` records.

### PR DESCRIPTION
**Pull request by tgxworld**
* Default retention of 90 days.

### CODE REVIEW

File: lib/email/receiver.rb [hunk 1]
```diff
@@ -171,8 +171,20 @@ def process_internal
 
         raise first_exception if first_exception
 
-        if post = find_related_post(force: true)
-          if Guardian.new(user).can_see_post?(post) && post.created_at < 90.days.ago
```
Comment (tgxworld) [inline lib/email/receiver.rb]: @ZogStriP I wanted to get your thoughts on this. If I'm understanding our requirements correctly, we only want to raise a `OldDestinationError` when the PostReplyKey is invalid. The assumption here is would mean that an invalid `PostReplyKey` means old though.

Comment (tgxworld) [reply lib/email/receiver.rb]: Hmm I realized that the assumption where if we can't find a `PostReplyKey` means that the destination is old. Reverted in https://github.com/discourse/discourse/pull/6289/files#diff-0327bfc5fb2dd2a7fd635a096e7fb0daR187

File: lib/email/receiver.rb [hunk 2]
```diff
@@ -171,8 +171,20 @@ def process_internal
...
+            all_destinations.each do |destination|
+              self.class.check_address(destination,
+                throw_on_invalid_post_reply_key: true
+              )
+            end
+
+            false
+          end
+
+          if is_invalid && post.created_at < SiteSetting.delete_post_reply_keys_after_days.days.ago
```
Comment (ZogStriP) [inline lib/email/receiver.rb]: What if `SiteSetting.delete_post_reply_keys_after_days == 0`?

File: config/locales/server.en.yml [hunk 3]
```diff
@@ -1565,6 +1565,7 @@ en:
     unsubscribe_via_email_footer: "Attach an unsubscribe via email mailto: link to the footer of sent emails"
 
     delete_email_logs_after_days: "Delete email logs after (N) days. 0 to keep indefinitely"
+    delete_post_reply_keys_after_days: "Delete post_reply_keys after (N) days. 0 to keep indefinitely"
```
Comment (ZogStriP) [inline config/locales/server.en.yml]: I would actually name that setting something like `allow_email_reply_for_days`. 

That is more user friendly than the `reply keys` technicality.

File: lib/email/receiver.rb (hunk 1 above)
Comment (ZogStriP) [reply lib/email/receiver.rb]: I don't think we need to specifically check for an invalid post reply key. If the user takes too much time to reply via email, just gently tell them to visit the forum.

File: config/locales/server.en.yml (hunk 3 above)
Comment (tgxworld) [reply config/locales/server.en.yml]: > That is more user friendly than the reply keys technicality. 
Good idea :+1:  Let me update the name for the site setting.

File: lib/email/receiver.rb (hunk 2 above)
Comment (tgxworld) [reply lib/email/receiver.rb]: ah good catch :+1: Let me fix :stuck_out_tongue_closed_eyes:

File: lib/email/receiver.rb (hunk 1 above)
Comment (tgxworld) [reply lib/email/receiver.rb]: @ZogStriP Yea I think we can try that too. I think I'm a little overzealous in getting it 100% accurate for post replies via email.

**Pull request merged**